jetbot/bot.py

```diff
@@ -451,48 +470,68 @@ async def channels(ctx: commands.Context):
                             category = await guild.create_category(item["category_name"])
                         categories[item["category_name"].lower().strip()] = category
 
-                    # Get the channel
-                    channel = await self.product_channel(item["channel_name"], item["category_name"])
-                    if not channel:
-                        print("Creating channel... " + item["channel_name"])
-                        channel = await guild.create_text_channel(item["channel_name"], category=category)
-                        new.append(channel)
-
-                    # Set permissions
-                    print("Updating channel... " + item["channel_name"] + " in " + item["category_name"])
-                    for overwrite in channel.overwrites:
-                        await channel.set_permissions(overwrite, overwrite=None)
-                    await channel.edit(slowmode_delay=5, sync_permissions=True)
-
-                    # Handle read-only
-                    if "read_only" in item and item["read_only"]:
-                        # Send the read-only message
-                        last = [f async for f in channel.history(limit=1)]
-                        if not last or last[0].content != item["read_only"]:
-                            await channel.send(item["read_only"])
-
-                        # Set the topic
+                    # Handle channels configuration
+                    if "channels" in item:
+                        for channel_config in item["channels"]:
+                            channel = None
+                            for existing in valid_channels:
+                                if existing.name == channel_config["name"] and existing.category and \
+                                   existing.category.name.lower().strip() == item["category_name"].lower().strip():
+                                    channel = existing
+
+                            # Reset permissions and set basic configuration
+                            print("Updating channel... " + channel_config["name"] + " in " + item["category_name"])
+                            for overwrite in channel.overwrites:
+                                await channel.set_permissions(overwrite, overwrite=None)
+                            await channel.edit(slowmode_delay=5, sync_permissions=True)
+
+                            # Handle channel type first
+                            if "type" in channel_config:
+                                if channel_config["type"] == "forum" and not isinstance(channel, ForumChannel):
+                                    await channel.edit(type=ChannelType.forum)
+                                elif channel_config["type"] == "stage_voice" and not isinstance(channel, StageChannel):
+                                    await channel.edit(type=ChannelType.stage_voice)
+
+                    channels_to_update = await self.product_channels(item)
```

Comment:
I'm confused why we're having to call out to `product_channels` here, when the lines above are locating all the existing channels (and should also be creating any new ones)?